### PR TITLE
fix: select: address selected items getting unselected when expanded.

### PR DESCRIPTION
**Pull request by ysarvaiya-c-eightfold**
## SUMMARY:
This fix addresses the issue of Selected list item getting unselected when expanded or focus is on select dropdown container.

when keydown is called when enter is pressed on select dropdown. onClick event listener for close button on the selected pill item is getting called causing the selected item to remove from the selected list.

Fixed the issue by adding if guard to call toggle function only when currently active element in document is contained to the target close button.

## JIRA TASK (Eightfold Employees Only):
https://eightfoldai.atlassian.net/browse/ENG-95192

## CHANGE TYPE:

- [x] Bugfix Pull Request
- [ ] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [x] I have added unittests for this change

## TEST PLAN:
- Pull the PR branch locally and run `yarn storybook`. 
- Verify the new added Form multiple select story on form and confirm selected items are not getting removed after expanding the select dropdown. (`http://localhost:2022/?path=/story/form--form-select-multiple`)

https://github.com/user-attachments/assets/a7c3ac19-7e23-4e8a-9307-e0769a4fa5d0

### CODE REVIEW

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -577,7 +578,11 @@ export const Select: FC<SelectProps> = React.forwardRef(
             disabled={mergedDisabled}
             key={`select-pill-${index}`}
             label={opt.text}
-            onClose={() => toggleOption(opt)}
+            onClose={(event) => {
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:581]: can you add unit test for the scenario?

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -577,7 +578,11 @@ export const Select: FC<SelectProps> = React.forwardRef(
...
+              if (contains(document.activeElement, event.target as Node)) {
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:582]: whats the idea here? why does it toggle option on close at all? seems like onclick/onkeydown should be solely responsible for toggling an option

File: src/components/Select/Select.tsx
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:1]: can you add a unit test for this behavior please

File: src/components/Select/Select.tsx (hunk 2 above)
Comment (jhoward-eightfold) [reply src/components/Select/Select.tsx:582]: I only just now realized that this is the pill onClose and not the dropdown on close... is there a possibly a propagation issue in play here?

File: src/components/Form/Form.stories.tsx
Comment (jhoward-eightfold) [inline src/components/Form/Form.stories.tsx:1]: pulled this locally and played around with it. It is having the desired outcoming, though I don't full understand why the bug is happenning

**Review by jhoward-eightfold — APPROVED** [review]

File: src/components/Select/Select.tsx (hunk 2 above)
Comment (ysarvaiya-c-eightfold) [reply src/components/Select/Select.tsx:582]: No. I’ve debugged the propagation part. event propagation is not affecting the onClose behavior for the pill component.

Comment (jhoward-eightfold) [reply src/components/Select/Select.tsx:582]: Do you have an understanding of how this is getting triggered?

**Pull request merged**